### PR TITLE
CA-182869: syntax extension for vm plugin call --- args:key:file=filename 

### PR DESCRIPTION
**Pull request by zli**

### CODE REVIEW

**Comment by jonludlam** [comment]
I'm not so keen on the special casing of the CLI argument handling, but on the other hand it's not totally clear what else we can do. Possibly we could have e.g. 'script:localfile=filename'? And then we can make it generic. Also we should update the documentation in cli_frontend at least.

**Comment by zli** [comment]
@jonludlam Update as discussed. Interestingly github shows `args:key:file` as emoji args:key:file

**Comment by simonjbeaumont** [comment]
>  Interestingly github shows `args:key:file` as emoji args:key:file

Brilliant!

File: ocaml/xapi/cli_operations.ml [hunk 1]
```diff
@@ -2040,12 +2040,25 @@ let vm_memory_target_wait printer rpc session_id params =
...
+let vm_call_plugin fd printer rpc session_id params =
 	let vm_uuid = List.assoc "vm-uuid" params in
 	let vm = Client.VM.get_by_uuid rpc session_id vm_uuid in
 	let plugin = List.assoc "plugin" params in
 	let fn = List.assoc "fn" params in
 	let args = read_map_params "args" params in
+	(* Syntax interpretation: args:key:file=filename equals args:key=filename_content *)
+	let convert ((k,v) as p) =
+		match String.split ~limit:2 ':' k with
+		| key :: annot :: [] when annot = "file" ->
```
Comment (johnelse) [inline ocaml/xapi/cli_operations.ml]: Since the `annot` symbol isn't used anywhere else, why not just match against `key :: "file" :: []` ?

Comment (zli) [inline ocaml/xapi/cli_operations.ml]: You are right, good suggestion! Updated now.

**Pull request merged**